tikv-client/src/main/java/com/pingcap/tikv/types/EnumType.java

```diff
@@ -92,9 +92,19 @@ private Integer parseEnumValue(Integer number) throws ConvertOverflowException {
   /** {@inheritDoc} */
   @Override
   protected Object decodeNotNull(int flag, CodecDataInput cdi) {
-    if (flag != Codec.UVARINT_FLAG)
-      throw new TypeException("Invalid EnumType(IntegerType) flag: " + flag);
-    int idx = (int) IntegerCodec.readUVarLong(cdi) - 1;
+    int idx;
+    switch (flag) {
+        // encodeKey write UINT_FLAG
+      case Codec.UINT_FLAG:
+        idx = (int) IntegerCodec.readULong(cdi) - 1;
+        break;
+        // encodeValue write UVARINT_FLAG
```

Comment:
`encodeValue` -> `decodeValue`

Reply:
What I want to comment here is that UVARINT_FLAG is used during encodeValue, so we need to handle it in decode